mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,623 @@
+        phase signals (first column of `ixs`).
+    [amp_signal] : array, shape (n_amp_signals, n_times,)
+        Only returned if `return_data` is True. The amplitude timeseries of the
+        amplitude signals (second column of `ixs`).
+
+    References
+    ----------
+    [1] This function uses the PacPy module developed by the Voytek lab.
+        https://github.com/voytekresearch/pacpy
+    """
+    from mne.io.base import _BaseRaw
+    if not isinstance(inst, _BaseRaw):
+        raise ValueError('Must supply Raw as input')
+    sfreq = inst.info['sfreq']
+    data = inst[:][0]
+    pac = _phase_amplitude_coupling(data, sfreq, f_phase, f_amp, ixs,
+                                    pac_func=pac_func, events=events,
+                                    tmin=tmin, tmax=tmax, n_cycles=n_cycles,
+                                    scale_amp_func=scale_amp_func,
+                                    return_data=return_data,
+                                    concat_epochs=concat_epochs,
+                                    n_jobs=n_jobs, verbose=verbose)
+    # Collect the data properly
+    if return_data is True:
+        pac, freq_pac, data_ph, data_am = pac
+        return pac, freq_pac, data_ph, data_am
+    else:
+        pac, freq_pac = pac
+        return pac, freq_pac
+
+
+def _phase_amplitude_coupling(data, sfreq, f_phase, f_amp, ixs,
+                              pac_func='ozkurt', events=None,
+                              tmin=None, tmax=None, n_cycles=3,
+                              scale_amp_func=None, return_data=False,
+                              concat_epochs=False, n_jobs=1,
+                              verbose=None):
+    """ Compute phase-amplitude coupling using pacpy.
+
+    Parameters
+    ----------
+    data : array, shape ([n_epochs], n_channels, n_times)
+        The data used to calculate PAC
+    sfreq : float
+        The sampling frequency of the data.
+    f_phase : array, dtype float, shape (n_bands, 2,)
+        The frequency range to use for low-frequency phase carrier.
+    f_amp : array, dtype float, shape (n_bands, 2,)
+        The frequency range to use for high-frequency amplitude modulation.
+    ixs : array-like, shape (n_pairs x 2)
+        The indices for low/high frequency channels. PAC will be estimated
+        between n_pairs of channels. Indices correspond to rows of `data`.
+    pac_func : {'plv', 'glm', 'mi_canolty', 'mi_tort', 'ozkurt'} |
+               list of strings
+        The function for estimating PAC. Corresponds to functions in
+        `pacpy.pac`. Defaults to 'ozkurt'.
+    events : array, shape (n_events, 3) | array, shape (n_events,) | None
+        MNE events array. To be supplied if data is 2D and output should be
+        split by events. In this case, `tmin` and `tmax` must be provided. If
+        `ndim == 1`, it is assumed to be event indices, and all events will be
+        grouped together.
+    tmin : float | list of floats, shape (n_pac_windows,) | None
```

Comment:
There was some naming inconsistency there. I'm going to rename the number of pac time windows (aka `len(tmin)`)  to be `n_pac_windows`, and will reserve `n_times` for reference to the timeseries data itself (e.g., `raw.n_times`)